spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Cell.scala

```diff
@@ -81,39 +81,39 @@ abstract class Cell[T : ClassTag](
    *      and recursively intialize all the tensors in the Table.
    *
    * @param hidden
-   * @param size batchSize
+   * @param batchSize batchSize
    * @return
    */
-  def hidResize(hidden: Activity, size: Int, rows: Int = 1, columns: Int = 1): Activity = {
+  def hidResize(hidden: Activity, batchSize: Int, imageSize: Array[Int] = null): Activity = {
     if (hidden == null) {
       if (hiddensShape.length == 1) {
-        hidResize(Tensor[T](), size, rows, columns)
+        hidResize(Tensor[T](), batchSize)
       } else {
         val _hidden = T()
         var i = 1
         while (i <= hiddensShape.length) {
           _hidden(i) = Tensor[T]()
           i += 1
         }
-        hidResize(_hidden, size, rows, columns)
+        hidResize(_hidden, batchSize, imageSize)
       }
     } else {
       if (hidden.isInstanceOf[Tensor[T]]) {
         require(hidden.isInstanceOf[Tensor[T]],
           "Cell: hidden should be a Tensor")
-        hidden.toTensor.resize(size, hiddensShape(0))
+        hidden.toTensor.resize(batchSize, hiddensShape(0))
       } else {
         require(hidden.isInstanceOf[Table],
           "Cell: hidden should be a Table")
         var i = 1
-        if (rows== 1 && columns==1) {
+        if (null == imageSize) {
           while (i <= hidden.toTable.length()) {
-            hidden.toTable[Tensor[T]](i).resize(size, hiddensShape(i - 1))
+            hidden.toTable[Tensor[T]](i).resize(batchSize, hiddensShape(i - 1))
             i += 1
           }
         } else {
           while (i <= hidden.toTable.length()) {
-            hidden.toTable[Tensor[T]](i).resize(size, hiddensShape(i - 1), rows, columns)
+            hidden.toTable[Tensor[T]](i).resize(Array(batchSize, hiddensShape(i - 1)) ++ imageSize)
```

Comment:
hidResize will be called for each input. Thus, ++ operation will new three arrays and trigger minor gc.

Can imageSize be a fixed dimension array?